### DISCUSSION TITLE
Git scraping support, thoughts?

### DISCUSSION THREAD

**Discussion by frosencrantz** [saulpw/visidata#1103] [Ideas]
Simon Wilson has been talking about a scraping technique he calls git scraping
where a person scrapes data from a website and saves the data to git.

Here are some link about the subject:
- https://simonwillison.net/2020/Oct/9/git-scraping/
- GitHub has some experimental support:
  https://next.github.com/projects/flat-data
- There is a topic: https://github.com/topics/git-scraping

VisiData could do a great job at supporting this data model. There would
need to be builtin support for understanding that there are different versions
of the same data file, with an ordering of those versions.  This might be based
on source code history, but could also be files in a directory based on
trailing digits or timestamps alone.  The UI might need to provide some changes
that would allow picking two (or more) versions for comparison or for joining.
It would be useful if VisiData could remember column types across versions, or
even transformations.  Maybe VisiData could assist if the format of the data changed
like new columns or renamed columns?

If VisiData understood the ordering  it might be possible to create a frequency sheet
that shows the difference between two or more versions.

**Comment by saulpw** [comment]
I could indeed see this as being great, @frosencrantz.  At some point I'll put some more work into vdplus/vgit and this could be a part of that.

**Comment by frosencrantz** [comment]
Sounds fantastic, @saulpw!

I'll mention:

- I've tried vgit a couple times, but it never worked for me.  I couldn't tell if it was taking a long time to load, or was just broken, and if that was my environment or something else.  Sorry not to report this.
- I do like the git aspect of this, but I also at times end up with version 1,...,N of a data file (think spreadsheet or other files) and it would be useful to be able to explore history with Visidata.
- On occasion, I've had to use another version control system, so any builtin flexibility would be useful.

**Reply by saulpw** [reply]
I agree, it would be great to explore history with VisiData.  I'd love to find someone who loves both git and VisiData and wants to help build the ultimate git explorer/editor.

**Comment by frosencrantz** [comment]
It looks like Simon Wilson has started to doing something along these lines. It is different than I was thinking.  I thought it might be useful to compare the different results.  It looks like he is more interested in providing all the data in one database that spans all versions.  That also seems useful.

https://simonwillison.net/2021/Dec/7/git-history/
